scripts/rosetta/test-block-race-with-data.sh

```diff
@@ -0,0 +1,309 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Script to test block race condition with real mainnet archive data
+# Sets up postgres database, imports dump, and runs the test
+
+# Default values
+PORT_BASE=17000
+POSTGRES_PORT=15000
+NUM_BLOCKS=3
+NUM_ZKAPP_TXS=10
+NUM_PAYMENTS=110
+
+# Data URLs
+ARCHIVE_DUMP_URL="https://storage.googleapis.com/mina-archive-dumps/mainnet-archive-dump-2025-11-11_0000.sql.tar.gz"
+LEDGER_URL="https://storage.googleapis.com/o1labs-ci-test-data/ledgers/single-bp-ledger.tar"
```

Comment:
I would be nice to have infromation how this was generated. In case we would need to update it in the future